WarOOP/Models/Warrior.cs

```diff
@@ -17,7 +17,22 @@ public Warrior()
         Attack = 5;
     }
 
-    private void GetDamageFrom(Warrior enemy) => CurrentHealth -= enemy.Attack;
+    protected virtual void GetDamageFrom(Warrior enemy)
+    {
+        if (IsAlive) {CurrentHealth -= enemy.Attack;}
+    }
 
-    public void AttackTo(Warrior enemy) => enemy.GetDamageFrom(this);
-}
+    public void AttackTo(Warrior enemy)
+    {
+        if (IsAlive) {enemy.GetDamageFrom(this);}
+    }
+
+    public static Warrior CreateWarrior(Type type)
```

Comment:
You can use generic type parameter and constraints to create new instance based on the type